src/bot/lib/crud.js

```diff
@@ -0,0 +1,32 @@
+// =============
+// CRUD HELPERS
+// =============
+
+// Add item (skills or categories) with a given name.
+export function addItemByName(token, name, queryFunction) {
+  return queryFunction({token, name});
+}
+
+// processes adds and returns output for dialogue
+export function prepareAddOutput(matchResults, token, queryFunction) {
+  const {search, matches, match, exact} = matchResults;
+  const output = [];
+
+  if (matches.length === 0) {
+    return addItemByName(token, search, queryFunction).then(results => {
+      output.push(`_You have successfully added "${search}"._`);
+      return {output};
+    });
+  }
+
+  // else suggest they use list
+  const itemsList = matches.map(item => item.name).join(', ');
+  output.push(`_You specified "${search}", which matches: *${itemsList}*. ` +
```

Comment:
The "You specified __, which matches: __" seems redundant, considering this code should ensure a similar message is sent to the user: https://github.com/bocoup/skillsbot/blob/2dd3b56/src/bot/lib/matching.js#L35-L49

Perhaps just display the "Try using the `list` command to see what items already exist." message?